### ISSUE TITLE
Question: Differential accuracy from 2 recievers

### ISSUE THREAD

**Issue by jgwinner** [tomojitakasu/RTKLIB#421]
Hello:

I have an application that's essential to detect the relative position and orientation from 2 receivers. (I can use a 3rd if necessary, that would be about 1m from one of the two receivers). 

Basically, it's a "Wand" that directs where a Rover moves to. BOTH will be moving. 

Can RTKLIB do this in RTK mode (vs PPK)?

Most applications care about absolute positioning; I don't care about that so so much, but need a dead accurate distance and orientation between the two stations.

I was going to use infrared and ultrasonics to do this, but they can have problems.

Hoping RTKLIB can solve this!

**Comment by yydgis** [comment]
This is called moving base RTK or heading determination. The rtklib
provides this option. If you purely want the heading between two antenna,
you can purchase the GPS receiver with heading option. Hope this helps.

**Comment by jgwinner** [comment]
Thanks.

How accurate is the delta _position_ between the two nodes with moving base RTK?

I can get orientation from IMU units easily, but it's critical for me to know if the other node is 'in front of' or 'behind' (etc) and the distance; I've read RTK can get down to a few CM and want to see if that's theoretical or if it's reasonable.

== John ==

**Comment by yydgis** [comment]
yes, the relatively position between two antennas is in cm level accuracy
with the RTK fix solution. I am a little curiosity about the "front vs.
behind" situation, the antennas are attached in the moving vehicle, you
already know the relative position of the two antennas, the "front vs
behind" will depend on your driving direction (i.e., from your IMU output).
right?>

**Comment by jgwinner** [comment]
It's a robot, that's being directed by a 'wand'. If the wand is pointed away from the robot, it shouldn't move. Only if the wand is pointed at the robot, and a button is depressed, should the robot move.

Another mode is pointing at the ground with the wand, pushing a different button, and having the robot navigate to the position on the ground, but this is "Phase 2".

No one is driving the robot. However, you can sit ON the robot and move it via the wand, if needed. This could be more of a "joystick" movement though so I didn't include that in the original description. 

== John ==

**Comment by jgwinner** [comment]
The idea was one GPS device in the wand, another in the robot.

**Comment by reid-p** [comment]
On 19/07/2018 04:43, John D. Gwinner wrote:
> The idea was one GPS device in the wand, another in the robot.
> 

I think moving base requires the location be ANT1 & ANT2 to be a fixed distance apart
I'd love to be wrong here as I've got a similar application where I need to
know the relative distance between two rovers, but absolute position
is not needed to cm level.


-- 
Regards
Phil Reid

**Comment by yydgis** [comment]
hi Phil
There is no assumption for the moving base about the fix distance. But I am
not sure that th code in rtklib made such assumption or not

Phil <notifications@github.com>于2018年7月18日 周三下午6:56写道：

**Comment by reid-p** [comment]
On 19/07/2018 12:19, dan wrote:
> hi Phil
> There is no assumption for the moving base about the fix distance. But I am
> not sure that th code in rtklib made such assumption or not
> 
G'day Dan,

So the "Baseline length Constraint" is optional.
I hadn't read the manual that way. Good news.

Thanks
Phil

**Comment by yydgis** [comment]
Yes, the baseline length constraint is optional, I do not remember that it
is default in the code or not. But in principle, moving base method works
without the length constraint.
